app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -105,14 +106,25 @@ class AuditController @Inject() (
       Future.successful(Accepted)
     }
 
-  def postStatusAudit(auditType: AuditType)(implicit request: Request[Source[ByteString, _]]): Future[Result] =
+  def postStatusAudit(auditType: AuditType)(implicit request: Request[Source[ByteString, _]]): Future[Result] = {
+
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+    val auditSource = request.headers.get(XAuditSourceHeader) match {
+      case Some(value) => value
+      case None        => auditType.source
+    }
```

Comment:
`request.headers.get(XAuditSourceHeader).getOrElse(auditType.source)` -- it's a minor thing but we shouldn't pattern match on things like this when it's unnecessary -- the extractor objects involved might make this more expensive.